packages/flutter/lib/src/cupertino/bottom_tab_bar.dart

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show ImageFilter;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+
+// Standard iOS 10 tab bar height.
+const double _kTabBarHeight = 50.0;
+
+const Color _kDefaultTabBarBackgroundColor = const Color(0xCCF8F8F8);
+
+class CupertinoTabBar extends StatelessWidget {
+  CupertinoTabBar({
+    Key key,
+    @required this.items,
+    this.onTap,
+    this.currentIndex: 0,
+    this.backgroundColor: _kDefaultTabBarBackgroundColor,
+    this.activeColor: CupertinoColors.activeBlue,
+    this.inactiveColor: CupertinoColors.inactiveGray,
+    this.iconSize: 24.0,
+  }) : super(key: key) {
+    assert(items != null);
+    assert(items.length >= 2);
+    assert(0 <= currentIndex && currentIndex < items.length);
+    assert(iconSize != null);
+  }
+
+  /// The interactive items laid out within the bottom navigation bar.
+  final List<BottomNavigationBarItem> items;
+
+  /// The callback that is called when a item is tapped.
+  ///
+  /// The widget creating the bottom navigation bar needs to keep track of the
+  /// current index and call `setState` to rebuild it with the newly provided
+  /// index.
+  final ValueChanged<int> onTap;
+
+  /// The index into [items] of the current active item.
+  final int currentIndex;
+
+  /// The background color of the tab bar. If it contains transparency, the
+  /// tab bar will automatically produce a blurring effect to the content
+  /// behind it.
+  final Color backgroundColor;
+
+  /// The foreground color of the icon and title for the [BottomNavigationBarItem]
+  /// of the selected tab.
+  final Color activeColor;
+
+  /// The foreground color of the icon and title for the [BottomNavigationBarItem]s
+  /// in the unselected state.
+  final Color inactiveColor;
+
+  /// The size of all of the [BottomNavigationBarItem] icons.
+  ///
+  /// This value is used to to configure the [IconTheme] for the navigation
+  /// bar. When a [BottomNavigationBarItem.icon] widget is not an [Icon] the widget
+  /// should configure itself to match the icon theme's size and color.
+  final double iconSize;
+
+  @override
+  Widget build(BuildContext context) {
+    final bool addBlur = backgroundColor.opacity < 1.0;
+
+    Widget returnWidget;
+
+    returnWidget = new DecoratedBox(
```

Comment:
prefer `result` for the return value variable.

prefer `Widget result = ...` so that the variable is never null, it'll make it easier for us to switch to non-nullable types later